### PR TITLE
Update influxdb client

### LINKED ISSUE #6088: Switch influxdb client dependency to use github.com/influxdata/influxdb1-client
@jefferai I saw you mention a dependency on github.com/influxdata/platform in https://github.com/golang/go/issues/29798, so I ran `go mod why`, from the `gomodtry2` branch, to see the source of the dependency:

```
# github.com/influxdata/platform
github.com/hashicorp/vault/plugins/database/influxdb
github.com/influxdata/influxdb/client/v2
github.com/influxdata/influxdb/models
github.com/influxdata/platform/models
```

We have recently split the client into its own self-contained package, https://github.com/influxdata/influxdb1-client, so you should depend on that instead to reduce the dependency bloat.

### PR DESCRIPTION
**Pull request by tsaarni**
Influxdb v1 client has been split into separate module from main influxdb code base. This change uses the correct client, which also allows us to get updates and avoids confusing some vuln scanners that flagged previous version incorrectly with vulnerabilities that applied only to the main code base.

Fixes #6088.

### CODE REVIEW

**Comment by tsaarni** [comment]
Note that this PR updates the current influxdb1 client to the correct library at https://github.com/influxdata/influxdb1-client. 

Issue #12230 requests influxdb2 support, which would be https://github.com/influxdata/influxdb-client-go instead.

**Comment by tsaarni** [comment]
Updated the PR with missing changelog entry.

**Comment by erikgb** [comment]
We are seeing our image scanner triggering on this. Any chance to get this PR rebased, reviewed and merged?

**Comment by tsaarni** [comment]
I've merged main. @pmmukh would you happen to know if this PR could be reviewed and considered for merging to get rid of security alerts? Should be relatively trivial.

**Review by erikgb — APPROVED** [review]
LGTM

**Comment by benashz** [comment]
Related tests are passing:

```

File: changelog/12262.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+secrets/database/influxdb: Upgrade influxdb1 client library and import it from new URI
```
Comment (benashz) [inline changelog/12262.txt:2]:
```suggestion
secrets/database/influxdb: Switch/upgrade to the `influxdb1-client` module
```

**Review by benashz — APPROVED** [review]
LGTM!

**Review by benashz — COMMENTED** [review]
Looking good. Only a small update to the changelog stub.

**Comment by benashz** [comment]
> After applying the changelog update `TestGauge_MaximumMeasurements` failed. Does not appear to be influxdb related, possibly flaky test?

Seems to be unrelated to this change. We'll look into thanks!

**Pull request merged**